### PR TITLE
fix for empty container formatting

### PR DESCRIPTION
**Pull request by Devansh-567**
The PR fixes the formatting of empty containers in `format_container()`.

Empty containers were printed as [] and(), while non-empty containers use spaces inside the delimiters, such as [ 1, 2, 3 ].

This change makes empty containers consistent:

* [] → [ ]
* () → ()

also added tests covering empty and non-empty vectors and sets.

### CODE REVIEW

**Review by fruffy — APPROVED** [review]

File: lib/log.h [hunk 1]
```diff
@@ -172,6 +172,10 @@ std::ostream &format_container(std::ostream &out, const Cont &container, char lb
             }
             out << Log::unindent;
         }
+    } else if (elems.empty()) {
+        // Keep empty containers consistent with the padded format used for
+        // non-empty containers.
+        out << lbrace << ' ' << rbrace;
```
Comment (ChrisDodd) [inline lib/log.h:178]: This seems deliberately wrong to me -- empty containres should NOT have spaces between the delimeters -- there should only be spaces around elements.

Comment (fruffy) [reply lib/log.h:178]: Seems like a style-thing. If we use `[ 1, 2, 3 ]` the empty list could also be `[ ]`. If we want `[]` then we should also print `[1, 2, 3]`, no?

Comment (ChrisDodd) [reply lib/log.h:178]: All other code (eg, json formatting) uses `[]` (no space) for empty vectors, with spaces (or newlines) around the elements for non-empty.  It even goes to extra effort to do that.

Comment (fruffy) [reply lib/log.h:178]: @Devansh-567 Can you do two things? 

Can you revert this PR according to @ChrisDodd's request. 

Separately, we should probably ensure that the compiler prints `[1, 2, 3]` consistently instead of using extra spaces. That aligns it more with the typical style. That should be a second, separate PR.

Comment (Devansh-567) [reply lib/log.h:178]: @fruffy, @ChrisDodd, thank you for the clarification. I'll open a PR right away per @ChrisDodd's point (agreed, it's inconsistent with how JSON output and other formatting in the codebase already handles this, []/() for empty, padding only around actual elements). Separately, I'll treat "make non-empty output [1, 2, 3] instead of [ 1, 2, 3 ]" as its own follow-up PR, not bundled into the revert.
